### DISCUSSION TITLE
Netbeans build infrastructure

### DISCUSSION THREAD

**Discussion by rostidev** [apache/netbeans#6006] [Q&A]
Hi,

I just looked at the code of Netbeans and have noticed that its build infrastructure is based on Ant. Why it's not based on a more modern Java build tool like Maven or Gradle instead or do you have some plan of switching into one of them sometime in a future?

**Comment by neilcsmith-net** [comment]
Well, one reason would be that earlier NetBeans developers were incapable of time travel! 😄 

There is a lot of legacy build code and custom build tasks. They work, they are well tested, and Ant is still maintained. The amount of work to migrate build system is not small, and should not be underestimated. Migration has been discussed before on the dev@ mailing list, which would also be a better place for this discussion. I would guess at this point the benefits don't outweigh the time and risks involved, but that's just my personal opinion.

Why do you ask?

**Comment by rostidev** [comment]
> Why do you ask?

Because I prefer Maven over Ant.

**Reply by neilcsmith-net** [reply]
> Because I prefer Maven over Ant.

So do I.  But the chances of switching to it for NetBeans is pretty minimal.  Gradle perhaps, but then that's just Ant with slightly better syntax. :wink: 

Actually have a look at everything that is in the build.  Or don't, as most people who contribute won't have to! :smile:
